cmd/ef_tests/test_runner.rs

```diff
@@ -46,10 +72,24 @@ pub fn parse_test_file(path: &Path) -> HashMap<String, TestUnit> {
     tests
 }
 
+fn validate_test(test: &TestUnit) {
+    // check that the decoded genesis block header matches the deserialized one
+    let genesis_rlp_as_string = test.genesis_rlp.clone();
+    let genesis_rlp_bytes = decode_hex(&genesis_rlp_as_string.clone()[2..]).unwrap();
+    let block_header = decode_block(&genesis_rlp_bytes).unwrap();
+    assert_eq!(block_header, test.genesis_block_header.clone().into());
+
+    // check that blocks can be decoded
+    for block in &test.blocks {
+        assert!(decode_block(block.rlp.as_ref()).is_ok() || block.expect_exception.is_some())
```

Comment:
you still have to assert that the decoded block is equal to the _core_ block. Additionally, I would check that encoding the _core_ block equals the rlp value. So that we test decoding & encoding.

Reply:
Done!